### ISSUE TITLE
eMMC hanging boot, no output over HDMI

### ISSUE THREAD

**Issue by jojo243** [ayufan-rock64/linux-build#183]
I got nearly the same issue as #172. I got a 4GB Rock64 and I used bionic-container, pre-release 0.6.36, booted from eMMC and upgraded to kernel 4.16. Now the board seems to hang and will neither boot nor show anything over HDMI. 

So my question is:
How do I "rescue" my eMMC? I don't have a eMMC/USB adapter and I was hoping to flash eMMC the same way I did it in the first place (boot from microSD with empty eMMC, eMMC was recognised and I flashed a fresh image release with dd). This doesn't seem to work though.

I can still boot from microSD with the jumper + eMMC in place (using [this](http://wiki.pine64.org/index.php/ROCK64_Software_Release#Xenial_Mate_Community_Build_Image_.5BmicroSD_to_eMMC.5D_.5B0.5.15-136.5D) image), but i never manage to see/mount eMMC filesystem inside the running system in order to restore a properly working image.
In other words
```bash
sudo fdisk -l
```
never shows `/dev/mmcblk0*`, only `/dev/mmcblk1*` which makes it impossible to write anything to eMMC. 

I tried several times to remove the jumper quickly, 2 to 3 seconds at power on after green LED lighting up, but then it doesn't boot at all nor is showing any output over HDMI.

Is it expected behaviour to see nothing happening via HDMI while in "boot loop"? Also, when I flashed SPI with the procedure described in https://github.com/ayufan-rock64/linux-build/blob/master/recipes/flash-spi.md, it doesn't show anything either (although it says so in the article). Does one need a serial console for this kind of output?
Is there anything I did/am doing wrong to flash the eMMC, maybe with the jumper?

**Comment by ayufan** [comment]
In 0.6.34 there is rock64_reset_emmc script that will init Emmc when it was
disabled before. This is to be used with jumper, after you remove it.

**Comment by jojo243** [comment]
Thanks for pointing that out. I will try the script and report back.
Would you generally recommend to flash images to eMMC using specially prepared images (e.g. [this one](http://wiki.pine64.org/index.php/ROCK64_Software_Release#Xenial_Mate_Community_Build_Image_.5BmicroSD_to_eMMC.5D_.5B0.5.15-136.5D)) or is it safe to boot from microSD with emmc+jumper and flash a fresh image with dd?

**Comment by ayufan** [comment]
Safe to boot and flash afterwards. I don’t like these images :)

**Comment by jojo243** [comment]
Nice! I don't like those either, I prefer your bleeding edge images :smile: Thx for quick response. The script seems to work.

Regarding HDMI output: I found the answer to my question [here](https://forum.pine64.org/showthread.php?tid=4924) (Scroll down to the "Background" section
>  The serial console is very important because the normal display does not show boot-up messages!

I had a little trouble getting the jumper removal thing right before the 3 seconds u-boot countdown, but in the end I did it by using another jumper (previously used a 6'' jumper wire with female connectors both ends, then switched to a standard jumper connector).
